### PR TITLE
[OCPCLOUD-1715]: Add exception annotations to mhc tests

### PR DESCRIPTION
**Pull request by odvarkadaniel**
This PR annotates all the `Expect` block with exceptions so we can easily trace bugs in the test suite.

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
I don't think we need to use so many format strings in these messages, the errors will be printed by the matcher anyway so including the error in the string in most of these cases is duplicating information in the logs, are we sure we want that? CC @elmiko

**Comment by elmiko** [comment]
good question @JoelSpeed , i did something similar in my changes as well. i tried running some tests with ginkgo/gomega to see what the errors looked like and it wasn't clear to me that they were printing as neatly as i liked, that's why i added it to the exception message.

i don't have strong feelings about it though, i do think we should agree on a general style for these.

**Comment by JoelSpeed** [comment]
My understanding is that the general format for the errors printed out would be
```
< stack to get to this failure>

< context message being added in this PR >

Expected <something>

<operator>

<the thing that didn't match>
```

So in the case of an error, it would be unexpected error, print errro after the context message, so I don't know we need to duplicate it within the context message, we certainly haven't where we have used this in CPMS

**Comment by elmiko** [comment]
> So in the case of an error, it would be unexpected error, print errro after the context message, so I don't know we need to duplicate it within the context message, we certainly haven't where we have used this in CPMS

sounds good to me, i'll update my patch as well

File: pkg/machinehealthcheck/machinehealthcheck.go [hunk 1]
```diff
@@ -41,31 +41,34 @@ var _ = Describe("MachineHealthCheck", framework.LabelMachineHealthChecks, func(
...
 
 		framework.WaitForMachineSet(client, machineSet.GetName())
 	})
 
 	AfterEach(func() {
 		specReport := CurrentSpecReport()
 		if specReport.Failed() {
-			Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed())
+			err := gatherer.WithSpecReport(specReport).GatherAll()
+			Expect(err).To(Succeed(), "failed to gather spec report")
```
Comment (JoelSpeed) [inline pkg/machinehealthcheck/machinehealthcheck.go:62]: Can we merge these back together? And the other cases where these have been split?

Comment (odvarkadaniel) [reply pkg/machinehealthcheck/machinehealthcheck.go:62]: Yup, sure. Good point.

Comment (odvarkadaniel) [reply pkg/machinehealthcheck/machinehealthcheck.go]: Reverted this change.

**Comment by JoelSpeed** [comment]
Looks good, are there any `Eventually` or `Consistently` assertions within this package, would be good to make sure those are annotated as well

**Comment by odvarkadaniel** [comment]
> Looks good, are there any `Eventually` or `Consistently` assertions within this package, would be good to make sure those are annotated as well

Nope. There are just `Expect` in this code. Checked earlier while working on this change. 
If you are ok with the annotations, this is good to go @JoelSpeed.

**Pull request merged**